src/software/simulation/simulator.h

```diff
@@ -64,30 +65,73 @@ class Simulator
      *
      * @param world The world to initialize the simulation with
      */
-    explicit Simulator(const World& world);
+    explicit Simulator(const Field& field);
     Simulator() = delete;
 
     /**
-     * Advances the simulation by the given time step. This will simulate
-     * physics and primitives
+     * Sets the state of the ball in the simulation. No more than 1 ball may exist
+     * in the simulation at a time. If a ball does not already exist, a ball
+     * is added with the given state. If a ball already exists, it's state is set to the
+     * given state.
      *
-     * @param time_step how much to advance the simulation by
+     * @param ball_state The new ball state
      */
-    void stepSimulation(const Duration& time_step);
+    void setBallState(const BallState& ball_state);
+
+    /**
+     * Removes the ball from the physics world
+     */
+    void removeBall();
+
+    /**
+     * Adds robots to the yellow team with the given initial states.
+     *
+     * @pre The robot IDs must not be duplicated and must not match the ID
+     * of any robot already on the yellow team.
+     *
+     * @throws runtime_error if any of the given robot ids are duplicated, or a
+     * yellow robot already exists with the ID
+     *
+     * @param robots the robots to add
+     */
+    void addYellowRobots(const std::vector<RobotStateWithId>& robots);
+
+    /**
+     * Adds robots to the yellow team with the given initial states.
+     *
+     * @pre The robot IDs must not be duplicated and must not match the ID
+     * of any robot already on the blue team.
+     *
+     * @throws runtime_error if any of the given robot ids are duplicated, or a
+     * yellow robot already exists with the ID
+     *
+     * @param robots the robots to add
+     */
+    void addBlueRobots(const std::vector<RobotStateWithId>& robots);
```

Comment:
über nit, can remove the documentation above and clump w/ addYellowRobots

Reply:
:+1: done